src/code/z_parameter.c

```diff
@@ -226,182 +229,195 @@ void func_80082644(PlayState* play, s16 alpha) {
         }
     } else {
         if (interfaceCtx->aAlpha != 255) {
-            interfaceCtx->aAlpha = alpha;
+            interfaceCtx->aAlpha = risingAlpha;
         }
     }
 }
 
-void func_8008277C(PlayState* play, s16 maxAlpha, s16 alpha) {
+/**
+ * Lower button alphas on the HUD to the requested value
+ * If (gSaveContext.hudVisibilityForceButtonAlphasByStatus), then instead update button alphas
+ * depending on button status
+ */
+void Interface_UpdateButtonAlphas(PlayState* play, s16 dimmingAlpha, s16 risingAlpha) {
     InterfaceContext* interfaceCtx = &play->interfaceCtx;
 
-    if (gSaveContext.unk_13E7 != 0) {
-        func_80082644(play, alpha);
+    if (gSaveContext.hudVisibilityForceButtonAlphasByStatus) {
+        Interface_UpdateButtonAlphasByStatus(play, risingAlpha);
```

Comment:
Ah right. Fig even has a video on it:
https://discord.com/channels/688807550715560050/838852326231769089/996940099549925457

There was a discussion about `hudVisibilityForceButtonAlphasByStatus`'s usage in EnMag and it's relation to z_demo.c. Discussion starts here:
https://discord.com/channels/688807550715560050/838852326231769089/996920043696828426

Reply:
tldr tho the usage in mag in a hack that is not at all related to its normal purpose, and should probably not be accounted for in its name. Its documented in my z_demo branch and it could possibly be noted in other places to make that clear as well

Reply:
I think leaning towards the dimming/raising approach for these two button alpha functions makes sense taking another look at it. Especially since the `raising` function is used in `Interface_Update` which is separate to this force-button-alpha member.

Reply:
I'd agree with this approach if the code was like
```c
if (hudVisibilityForceRisingButtonAlphas)
    Interface_RaiseButtonAlphas()
else
    Interface_DimButtonAlphas()
```
, but it is
```c
Interface_DimButtonAlphas()
{
    if (hudVisibilityForceRisingButtonAlphas)
        Interface_RaiseButtonAlphas()
        early return
    do dimming stuff
}
```
so calling Dim() can result in Raise() which sounds dumb

also Raise() can either raise or dim, and depends on button status, so Raise() doesn't seem descriptive enough

------------

However since `hudVisibilityForceRisingButtonAlphas` has a super niche use (minigames?, fishing, unrelated en_mag hack), (and as an aside I'd rather avoid delaying this PR),
I can get behind "whatever"

Reply:
`hudVisibilityForceRisingButtonAlphas` might as well be shortened to `forceRisingButtonAlphas` imo, especially since it's quite verbose rn, but I guess it's okay either way

Reply:
I think itd be fine as well to shorten it. A biref comment beside the variable explaining its higher level purpose can mention its for hud visibility. And its so specific that leaving out "hud visiblity" from the name wont conflict with other things

Reply:
I think I'm also fine with the shorter name. The common `hudVisibility` prefix was nice, but it's specific enough that it should be fine